gk/main.c

```diff
@@ -648,6 +576,143 @@ alloc_gk_conf(void)
 	return rte_calloc("gk_config", 1, sizeof(struct gk_config), 0);
 }
 
+static int
+get_ip_type(const char *ip_addr)
+{
+	int ret;
+	struct addrinfo hint;
+	struct addrinfo *res = NULL;
+
+	memset(&hint, 0, sizeof(hint));
+
+	hint.ai_family = PF_UNSPEC;
+	hint.ai_flags = AI_NUMERICHOST;
+
+	ret = getaddrinfo(ip_addr, NULL, &hint, &res);
+    	if (ret) {
+        	RTE_LOG(ERR, GATEKEEPER,
+			"gk: invalid ip address %s; %s\n",
+			ip_addr, gai_strerror(ret));
+        	return 1;
+    	}
+
+    	if(res->ai_family != AF_INET && res->ai_family != AF_INET6)
+		RTE_LOG(ERR, GATEKEEPER,
+			"gk: %s is an is unknown address format %d\n",
+			ip_addr, res->ai_family);
+
+	ret = res->ai_family;
+   	freeaddrinfo(res);
+
+	return ret;
+}
+
+int
+lua_init_tunnel(struct gk_config *gk_conf, const char *gw,
+	const char **gk_addrs, uint8_t num_gk_addrs,
+	const char **gt_addrs, uint8_t num_gt_addrs)
+{
+	int i;
+	int ret;
+	struct ether_addr gateway;
+
+	memset(&gk_conf->gk_inaddr, 0, sizeof(struct in_addr));
+	memset(&gk_conf->gk_inaddr6, 0, sizeof(struct in6_addr));
+
+	if (sscanf(gw, "%hhx:%hhx:%hhx:%hhx:%hhx:%hhx",
+           		&gateway.addr_bytes[0],
+           		&gateway.addr_bytes[1],
+           		&gateway.addr_bytes[2],
+           		&gateway.addr_bytes[3],
+           		&gateway.addr_bytes[4],
+           		&gateway.addr_bytes[5]) != sizeof(struct ether_addr)) {
+    		RTE_LOG(ERR, GATEKEEPER,
+			"gk: cannot parse gateway MAC address %s\n",
+			gw);
+		ret = -1;
+		goto out;
+	}
+
+	for (i = 0; i < num_gk_addrs; i++) {
+		int gk_type = get_ip_type(gk_addrs[i]);
+		if (gk_type == AF_INET) {
+			if ((ret = inet_pton(AF_INET, gk_addrs[i],
+					&gk_conf->gk_inaddr)) != 1) {
```

Comment:
Is the assignment to `ret` needed here? Since `ret` is changed to be -1 on failure anyway. There are three more instances of this below as well.